src/decode.rs

```diff
@@ -4851,211 +4851,152 @@ unsafe fn decode_b(
-                    *t.cur_sb_cdef_idx_ptr as libc::c_int,
-                    ts.msac.rng,
-                );
+
+            if DEBUG_BLOCK_INFO(f, t) {
+                println!("Post-cdef_idx[{}]: r={}", *t.cur_sb_cdef_idx_ptr, ts.msac.rng);
             }
         }
     }
-    if t.bx & 31 as libc::c_int >> ((*f.seq_hdr).sb128 == 0) as libc::c_int == 0
-        && t.by & 31 as libc::c_int >> ((*f.seq_hdr).sb128 == 0) as libc::c_int
-            == 0
-    {
-        let prev_qidx: libc::c_int = ts.last_qidx;
-        let have_delta_q: libc::c_int = ((*f.frame_hdr).delta.q.present != 0
-            && (bs as libc::c_uint
-                != (if (*f.seq_hdr).sb128 != 0 {
-                    BS_128x128 as libc::c_int
-                } else {
-                    BS_64x64 as libc::c_int
-                }) as libc::c_uint || b.skip == 0)) as libc::c_int;
-        let mut prev_delta_lf: [int8_t; 4] = [0; 4];
-        memcpy(
-            prev_delta_lf.as_mut_ptr() as *mut libc::c_void,
-            (ts.last_delta_lf).as_mut_ptr() as *const libc::c_void,
-            4 as libc::c_int as libc::c_ulong,
-        );
-        if have_delta_q != 0 {
-            let mut delta_q: libc::c_int = dav1d_msac_decode_symbol_adapt4(
+
+    // delta-q/lf
+    let shfl = ((*f.seq_hdr).sb128 == 0) as libc::c_int;
+    if t.bx & (31 >> shfl) == 0 && t.by & (31 >> shfl) == 0 {
+        let prev_qidx = ts.last_qidx;
+        let have_delta_q =
+            (*f.frame_hdr).delta.q.present != 0
+            && (
+                bs != (if (*f.seq_hdr).sb128 != 0 { BS_128x128 } else { BS_64x64 })
+                || b.skip == 0
+            );
+        let mut prev_delta_lf = ts.last_delta_lf;
+
+        if have_delta_q {
+            let mut delta_q = dav1d_msac_decode_symbol_adapt4(
                 &mut ts.msac,
                 (ts.cdf.m.delta_q.0).as_mut_ptr(),
-                3 as libc::c_int as size_t,
+                3,
             ) as libc::c_int;
-            if delta_q == 3 as libc::c_int {
-                let n_bits: libc::c_int = (1 as libc::c_int as libc::c_uint)
-                    .wrapping_add(
-                        dav1d_msac_decode_bools(
-                            &mut ts.msac,
-                            3 as libc::c_int as libc::c_uint,
-                        ),
-                    ) as libc::c_int;
-                delta_q = (dav1d_msac_decode_bools(
+
+            if delta_q == 3 {
+                let n_bits = 1 + dav1d_msac_decode_bools(&mut ts.msac, 3);
+                delta_q = dav1d_msac_decode_bools(
                     &mut ts.msac,
-                    n_bits as libc::c_uint,
-                ))
-                    .wrapping_add(1 as libc::c_int as libc::c_uint)
-                    .wrapping_add(((1 as libc::c_int) << n_bits) as libc::c_uint)
-                    as libc::c_int;
+                    n_bits
+                ) as libc::c_int + 1 + (1 << n_bits);
```

Comment:
I think this used to be unsigned adds (the `.wrapping_add`s).  I think we can move the `as libc::c_int` to around the whole result (`delta_q`), since `n_bits` is also unsigned here.